viam-cartographer.go

```diff
@@ -460,86 +401,70 @@ func (slamSvc *builtIn) GetInternalState(ctx context.Context, name string) ([]by
-func (slamSvc *builtIn) GetPointCloudMapStream(ctx context.Context, name string) (func() ([]byte, error), error) {
-	ctx, span := trace.StartSpan(ctx, "slam::builtIn::GetPointCloudMapStream")
+func (cartoSvc *cartographerService) GetPointCloudMapStream(ctx context.Context, name string) (func() ([]byte, error), error) {
+	ctx, span := trace.StartSpan(ctx, "viamcartographer::cartographerService::GetPointCloudMapStream")
 	defer span.End()
 
-	return grpchelper.GetPointCloudMapStreamCallback(ctx, name, slamSvc.clientAlgo)
+	return grpchelper.GetPointCloudMapStreamCallback(ctx, name, cartoSvc.clientAlgo)
 }
 
 // GetInternalStateStream creates a request, calls the slam algorithms GetInternalStateStream endpoint and returns a callback
 // function which will return the next chunk of the current internal state of the slam algo.
-func (slamSvc *builtIn) GetInternalStateStream(ctx context.Context, name string) (func() ([]byte, error), error) {
-	ctx, span := trace.StartSpan(ctx, "slam::builtIn::GetInternalStateStream")
+func (cartoSvc *cartographerService) GetInternalStateStream(ctx context.Context, name string) (func() ([]byte, error), error) {
+	ctx, span := trace.StartSpan(ctx, "viamcartographer::cartographerService::GetInternalStateStream")
 	defer span.End()
 
-	return grpchelper.GetInternalStateStreamCallback(ctx, name, slamSvc.clientAlgo)
+	return grpchelper.GetInternalStateStreamCallback(ctx, name, cartoSvc.clientAlgo)
 }
 
 // New returns a new slam service for the given robot.
-func New(ctx context.Context, deps registry.Dependencies, config config.Service,
-	logger golog.Logger, bufferSLAMProcessLogs bool,
+func New(
+	ctx context.Context,
+	deps registry.Dependencies,
+	config config.Service,
+	logger golog.Logger,
+	bufferSLAMProcessLogs bool,
 ) (slam.Service, error) {
-	ctx, span := trace.StartSpan(ctx, "slam::slamService::New")
+	ctx, span := trace.StartSpan(ctx, "viamcartographer::slamService::New")
 	defer span.End()
 
 	svcConfig, ok := config.ConvertedAttributes.(*slamConfig.AttrConfig)
 	if !ok {
 		return nil, rdkutils.NewUnexpectedTypeError(svcConfig, config.ConvertedAttributes)
 	}
 
-	primarySensorName, cams, err := configureCameras(ctx, svcConfig, deps, logger)
+	primarySensorName, cams, err := configureCameras(svcConfig, deps, logger)
 	if err != nil {
 		return nil, errors.Wrap(err, "configuring camera error")
 	}
 
-	modelName := string(config.Model.Name)
-	slamLib, ok := slam.SLAMLibraries[modelName]
-	if !ok {
-		return nil, errors.Errorf("%v algorithm specified not in implemented list", modelName)
-	}
-
-	slamMode, ok := slamLib.SlamMode[svcConfig.ConfigParams["mode"]]
-	if !ok {
-		return nil, errors.Errorf("getting data with specified algorithm %v, and desired mode %v",
-			modelName, svcConfig.ConfigParams["mode"])
+	subAlgo := SubAlgo(svcConfig.ConfigParams["mode"])
+	if subAlgo != dim2d {
+		return nil, errors.Errorf("%v does not have a subAlgo %v",
```

Comment:
[nit] The error message should say 'mode' instead of 'subAlgo', since 'mode' is the config parameter name.